lte/gateway/python/magma/pipelined/app/gy.py

```diff
@@ -229,6 +294,185 @@ def _install_default_flows_if_not_installed(self, datapath,
+                    "Failed to get flow msg '%s' for subscriber %s: %s",
+                    rule.id, imsi, err)
+                raise err
+        return flow_adds
+
+    def _get_classify_rule_flow_msgs(self, imsi, ip_addr, apn_ambr, flow, rule_num,
+                                     priority, qos, hard_timeout, rule_id,
+                                     app_name, app_service_type):
+        """
+        Install a flow from a rule. If the flow action is DENY, then the flow
+        will drop the packet. Otherwise, the flow classifies the packet with
+        its matched rule and injects the rule num into the packet's register.
+        """
+        flow_match = flow_match_to_magma_match(flow.match, ip_addr)
+        flow_match.imsi = encode_imsi(imsi)
+        flow_match_actions, instructions = self._get_classify_rule_of_actions(
+            flow, rule_num, imsi, ip_addr, apn_ambr, qos, rule_id)
+        msgs = []
+        if app_name:
+            # We have to allow initial traffic to pass through, before it gets
+            # classified by DPI, flow match set app_id to unclassified
+            flow_match.app_id = UNCLASSIFIED_PROTO_ID
+            parser = self._datapath.ofproto_parser
+            passthrough_actions = flow_match_actions + \
+                [parser.NXActionRegLoad2(dst=SCRATCH_REGS[1],
+                                         value=IGNORE_STATS)]
+            msgs.append(
+                flows.get_add_resubmit_current_service_flow_msg(
+                    self._datapath,
+                    self.tbl_num,
+                    flow_match,
+                    passthrough_actions,
+                    hard_timeout=hard_timeout,
+                    priority=self.UNCLASSIFIED_ALLOW_PRIORITY,
+                    cookie=rule_num,
+                    resubmit_table=self._enforcement_stats_scratch)
+            )
+            flow_match.app_id = get_app_id(
+                PolicyRule.AppName.Name(app_name),
+                PolicyRule.AppServiceType.Name(app_service_type),
+            )
+
+        if flow.action == flow.DENY:
+            msgs.append(flows.get_add_drop_flow_msg(
+                self._datapath,
+                self.tbl_num,
+                flow_match,
+                flow_match_actions,
+                hard_timeout=hard_timeout,
+                priority=priority,
+                cookie=rule_num)
+            )
+        else:
+            msgs.append(flows.get_add_resubmit_current_service_flow_msg(
+                self._datapath,
+                self.tbl_num,
+                flow_match,
+                flow_match_actions,
+                instructions=instructions,
+                hard_timeout=hard_timeout,
+                priority=priority,
+                cookie=rule_num,
+                resubmit_table=self._enforcement_stats_scratch)
+            )
+        return msgs
+
+    def _get_classify_rule_of_actions(self, flow, rule_num, imsi, ip_addr,
```

Comment:
Same for this, can we just put qos manager and version mapper as args and move to a common file?